Add papers to be used in the automatic language box paper

ptersilie opened the pull request.

ltratt commented on softdev.bib:
```diff
@@ -627,6 +627,30 @@ @inproceedings{jalili82building
 }
 
+@article{larcheveque95optimal,
+    author = {Larchevêque, J.-M.},
```
Can we get his/her full name?

ltratt commented on softdev.bib:
```diff
@@ -627,6 +627,30 @@ @inproceedings{jalili82building
+    pages = {1--15},
+    title = {Optimal incremental parsing},
+    volume = {17},
+    booktitle = {ACM Trans. Program. Lang. Syst.},
```
Shouldn't this be something like `journal = {TOPLAS}`?

ltratt commented on softdev.bib:
```diff
@@ -627,6 +627,30 @@ @inproceedings{jalili82building
+
+@article{petrone95reusing,
+    author = "Petrone, Luigi",
+    editor = "Thiagarajan, P. S.",
```
We don't include editor fields.

ltratt replied:
Unless this is a chapter in a book?

ptersilie replied:
Yes, looks like this is a book.

ptersilie replied:
https://link.springer.com/chapter/10.1007/3-540-60692-0_44

ltratt replied:
These look like conference proceedings to me (FSTTCS), so this should be `inproceedings`?

ptersilie replied:
Fixed.

ltratt commented on softdev.bib:
```diff
@@ -688,6 +712,23 @@ @phdthesis{visser97sdf
     school = {University of Amsterdam}
 }
 
+@book{visser97scannerless,
```
Indentation is wrong. And I think we have this already (and it's a phdthesis)?

ltratt replied:
Something is wrong with this entry. Is it a technical report? If so, it should have a number. Is it a PhD thesis? If so, we already have an entry claiming to be Eelco's PhD thesis, so at most one of those can be correct.

ptersilie replied:
Sorry, didn't see this. This seems to be a report. I'll update it.

ltratt commented on softdev.bib:
```diff
@@ -688,6 +712,23 @@ @phdthesis{visser97sdf
+@inproceedings{vandenbrand02disambiguation,
+  title = {Disambiguation filters for scannerless generalized LR parsers},
+  author = {Van den Brand, Mark GJ and Scheerder, Jeroen and Vinju, Jurgen J and Visser, Eelco},
+  booktitle = {International Conference on Compiler Construction},
```
This can just be `CC` for the conference.

ltratt commented on softdev.bib:
```diff
@@ -688,6 +712,23 @@ @phdthesis{visser97sdf
+  pages = {143--158},
+  year = {2002},
+  month = Mar,
+  organization = {Springer}
```
Organization? I don't think we need this field.

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
 }
 
+@inproceedings{fischer84poe,
+  title = {The Poe language-based editor project},
```
This should be `"The {Poe} language-based editor project"` I think.

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
 
+@inproceedings{fischer84poe,
+  title = {The Poe language-based editor project},
+  author = {Fischer, Charles N and Johnson, Gregory F and Mauney, Jon and Pal, Anil and Stock, Daniel L},
```
Should there be dots after the middle initials? Or just drop them?

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+@inproceedings{fischer84poe,
+  title = {The Poe language-based editor project},
+  author = {Fischer, Charles N and Johnson, Gregory F and Mauney, Jon and Pal, Anil and Stock, Daniel L},
+  booktitle = {ACM SIGSOFT Software Engineering Notes},
```
Is this a journal? Does it have an abbreviation? If so, let's use it.

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+@article{fischer92aladin,
+  title = {ALADIN: A scanner generator for incremental programming environments},
+  author = {Fischer, Bernd and Hammer, Carsten and Struckmann, Werner},
+  journal = {Software: Practice and Experience},
```
`SPE`

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+@article{bahlke86psg,
+  title = {The PSG system: from formal language definitions to interactive programming environments},
+  author = {Bahlke, Rolf and Snelting, Gregor},
+  journal = {ACM Transactions on Programming Languages and Systems (TOPLAS)},
```
`TOPLAS`

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+}
+
+@article{bahlke86psg,
+  title = {The PSG system: from formal language definitions to interactive programming environments},
```
"PSG" should be curly bracketed I think.

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+}
+
+@article{fischer92aladin,
+  title = {ALADIN: A scanner generator for incremental programming environments},
```
ALADIN should be curly bracketed.

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+}
+
+@article{ballance92pan,
+  title = {The Pan language-based editing system},
```
Pan should be curly bracketed.

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+@article{ballance92pan,
+  title = {The Pan language-based editing system},
+  author = {Ballance, Robert A and Graham, Susan L and Van De Vanter, Michael L},
+  journal = {ACM Transactions on Software Engineering and Methodology (TOSEM)},
```
`TOSEM`

ltratt commented on softdev.bib:
```diff
@@ -1769,6 +1810,54 @@ @techreport{fischer79locally
+
+@article{ballance92pan,
+  title = {The Pan language-based editing system},
+  author = {Ballance, Robert A and Graham, Susan L and Van De Vanter, Michael L},
```
Same thing about middle initials as earlier.

ltratt commented on softdev.bib:
```diff
@@ -1919,6 +2008,26 @@ @inproceedings{vanter00displaying
+@inproceedings{vanter94practical,
+  title = {Practical language-based editing for software engineers},
+  author = {Van De Vanter, Michael L},
+  booktitle = {{W}orkshop on Software Engineering and Human-Computer Interaction},
```
Does that workshop have an abbreviation? If so let's use it. Otherwise, let's drop the curly brackets around "W".

ltratt replied:
Ah, we're not supposed to put "workshop" in the title. It should just be "Software Engineering and Human-Computer Interaction" (or, preferably, if that workshop has an abbreviation, just use that).

ptersilie replied:
I see. I fixed it by using the abbreviation.

ltratt commented on softdev.bib:
```diff
@@ -1919,6 +2008,26 @@ @inproceedings{vanter00displaying
+@incollection{lewis95designing,
+  title = {Designing for error},
+  author = {Lewis, Clayton and Norman, Donald A},
+  booktitle = {Readings in Human--Computer Interaction},
```
I wonder if this has an abbreviation?

ltratt commented:
Sorry about this -- there are a few comments!

ptersilie commented:
Addressed your comments. I'm still unsure why I got `fires a naughty regex '.*[Ww]orkshop'` for the workshop paper, but this was solved by using the abbreviation now.

ltratt commented:
What about https://github.com/softdevteam/softdevbib/pull/47#discussion_r250161780?

ptersilie commented:
Sorry, I missed that as I marked it as resolved which hides it from this view (maybe I shouldn't do this). I fixed it as well now.

ltratt commented on softdev.bib:
```diff
@@ -711,11 +711,12 @@ @phdthesis{visser97sdf
+    author = {Visser, Eelco and others},
+    number = {P9707}
+    year = {1997},
+    publisher = {Universiteit van Amsterdam. Programming Research Group}
```
Is the publisher actually the PRG? Or is it a university wide technical report? [Typically it's either university wide or department wide. I've never seen it be research group wide, but maybe that's just me leading a sheltered life!]

ptersilie replied:
Really hard to tell. The only info I found is this: https://www.narcis.nl/publication/RecordID/oai:dare.uva.nl:publications%2F15977da5-73a9-4d9a-b7d9-f215630dc31d

ltratt replied:
I can't find anything obvious either. I can find a paper with Eelco as an author where he uses the same publisher as you do -- perhaps that's as good as we'll manage! Let's stick with this then.

ltratt commented:
Edd and I ended up deciding that the "resolved" thing is more trouble than it's worth, so we've stopped using it :/

ptersilie commented:
I use it to keep track of which issues I've resolved. I feel like it should auto unresolve once you add another comment to it. But you're right. Seems more trouble than it's worth.

ltratt commented:
There are a couple of errors to fix.

ptersilie commented:
Whoops, I only checked with `bin/prebib softdev.bib` (instead of `bin/prebib softdev.bib > /dev/null`) which prints out the result and thus hides the warnings at the beginning (I was expecting it to fail). Fixed now.

ptersilie commented:
Done! Who'd have thought that creating a good bibliography is so hard. ;-)

ltratt commented:
Can we change the commit title from "Add papers to be used in the automatic language box paper" to something like "Add various papers on <topic X>" where I think "<topic X>" is probably "parsing"?

ltratt commented:
[We want this to be a generically available repo, so we don't want people to think it's just for our papers.]

ptersilie commented:
Done.
